src/Provider.js

```diff
@@ -1,74 +1,47 @@
-import { Children, Component } from "react"
-import { polyfill } from "react-lifecycles-compat"
-import * as PropTypes from "./propTypes"
+import { Children, Component, createContext, createElement } from "react"
+import { shallowEqual } from "./utils/utils"
 
 const specialReactKeys = { children: true, key: true, ref: true }
 
-class Provider extends Component {
-    static contextTypes = {
-        mobxStores: PropTypes.objectOrObservableObject
-    }
+export const MobXProviderContext = createContext({})
```

Comment:
No, want to keep `Provider / inject` as is. I think `React.createContext` is the way to go now. But I do want to keep supporting for `Provider / inject` at least in v6 to not make the migration harder. (would be a shame if people wouldn't migrate to hooks, because they lack the time to factor Provider out)